lib/iris/cube.py

```diff
@@ -3021,8 +3021,7 @@ def __eq__(self, other):
             # having checked everything else, check approximate data
             # equality - loading the data if has not already been loaded.
```

Comment:
If we're happy with using `da.allclose`, I guess this comment about loading the data can be removed?

Reply:
If this is using 'self.data', 'other.data' then of course it is still realising the cubes.

In fact I don't think we can use dask here, as the result needs to be a concrete boolean, whereas da.allclose returns a lazy scalar value, even when the args are concrete ...
```
>>> da.allclose(np.array([1,2]), np.array([1,1]))
dask.array<all-aggregate, shape=(), dtype=bool, chunksize=()>
>>> _.compute()
False
>>> 
```

Reply:
> If this is using 'self.data', 'other.data' then of course it is still realising the cubes.

Ah yes of course, you'd want to use `core_data()` or `lazy_data()`

Definitely in learning mode here...

Is there no advantage to using `da.allclose` with the `compute` call?  I was thinking it would access the files in chunks and therefore avoid busting memory.

Aside: that thing you did with the underscore, I did not know was a thing :open_mouth: .

Reply:
>  you'd want to use core_data() 

I think it makes sense to use 'core_data' as it involves the minimum of extra work.
My key point was : the routine itself *must* still return a regular boolean, not a lazy value.
... but I see now you have just added the ".compute()" above ! :+1: 

> the underscore

Yes, it's where the interpreter puts unassigned results (in addition to printing them).
Also very handy when you forgot to save the result of a calculation !
```
>>> long_and_complicated()
BigObject(...)
>>> keep_that = _
```